fastify.d.ts

```diff
@@ -261,7 +261,9 @@ declare namespace fastify {
      * Registers a listener function that is invoked when all the plugins have
      * been loaded. It receives an error parameter if something went wrong.
      */
-    ready(readyListener?: () => void): void
+    ready(readyListener?: (err: Error) => void): void
+    ready(readyListener?: (err: Error, done: Function) => void): void
+    ready(readyListener?: (err: Error, context: FastifyInstance<HttpServer, HttpRequest, HttpResponse>, done: Function) => void): void
```

Comment:
If called without the `readyListener` parameter, a `Promise` that resolves to the `context` is returned.